pkg/tsdb/postgres/tlsmanager.go

```diff
@@ -0,0 +1,228 @@
+
+	settings.ConfigurationMethod = strings.TrimSpace(
+		strings.ToLower(datasource.JsonData.Get("tlsConfigurationMethod").MustString("file-path")))
+
+	if settings.ConfigurationMethod == "file-content" {
+		if err := m.writeCertFiles(datasource, &settings); err != nil {
+			return settings, err
+		}
+	} else {
+		settings.RootCertFile = datasource.JsonData.Get("sslRootCertFile").MustString("")
+		settings.CertFile = datasource.JsonData.Get("sslCertFile").MustString("")
+		settings.CertKeyFile = datasource.JsonData.Get("sslKeyFile").MustString("")
+		if err := validateCertFunc(settings.RootCertFile, settings.CertFile, settings.CertKeyFile); err != nil {
+			return settings, err
+		}
+	}
+	return settings, nil
+}
+
+type certFileType int
+
+const (
+	rootCert = iota
+	clientCert
+	clientKey
+)
+
+func (t certFileType) String() string {
+	switch t {
+	case rootCert:
+		return "root certificate"
+	case clientCert:
+		return "client certificate"
+	case clientKey:
+		return "client key"
+	default:
+		panic(fmt.Sprintf("Unrecognized certFileType %d", t))
+	}
+}
+
+func getFileName(dataDir string, fileType certFileType) string {
+	var filename string
+	switch fileType {
+	case rootCert:
+		filename = "root.crt"
+	case clientCert:
+		filename = "client.crt"
+	case clientKey:
+		filename = "client.key"
+	default:
+		panic(fmt.Sprintf("unrecognized certFileType %s", fileType.String()))
+	}
+	generatedFilePath := filepath.Join(dataDir, filename)
+	return generatedFilePath
+}
+
+// writeCertFile writes a certificate file.
+func writeCertFile(
+	ds *models.DataSource, logger log.Logger, fileContent string, generatedFilePath string) error {
+	fileContent = strings.TrimSpace(fileContent)
+	if fileContent != "" {
+		logger.Debug("Writing cert file", "path", generatedFilePath)
+		if err := ioutil.WriteFile(generatedFilePath, []byte(fileContent), 0600); err != nil {
+			return err
+		}
+		// Make sure the file has the permissions expected by the Postgresql driver, otherwise it will bail
+		if err := os.Chmod(generatedFilePath, 0600); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	logger.Debug("Deleting cert file since no content is provided", "path", generatedFilePath)
+	exists, err := fs.Exists(generatedFilePath)
+	if err != nil {
+		return err
+	}
+	if exists {
+		if err := os.Remove(generatedFilePath); err != nil {
+			return fmt.Errorf("failed to remove %q: %w", generatedFilePath, err)
+		}
+	}
+	return nil
+}
+
+func (m *tlsManager) writeCertFiles(ds *models.DataSource, settings *tlsSettings) error {
+	m.logger.Debug("Writing TLS certificate files to disk")
+	decrypted := ds.DecryptedValues()
+	tlsRootCert := decrypted["tlsCACert"]
+	tlsClientCert := decrypted["tlsClientCert"]
+	tlsClientKey := decrypted["tlsClientKey"]
+
+	if tlsRootCert == "" && tlsClientCert == "" && tlsClientKey == "" {
+		m.logger.Debug("No TLS/SSL certificates provided")
+	}
+
+	// Calculate all files path
+	workDir := filepath.Join(m.dataPath, "tls", ds.Uid+"generatedTLSCerts")
```

Comment:
Better to use id here instead of uid since uid can be changed.

```suggestion
	workDir := filepath.Join(m.dataPath, "tls", fmt.Sprintf("%dgeneratedTLSCerts", ds.Id))
```